supabase/migrations/20240313073422_init.sql

```diff
@@ -0,0 +1,1027 @@
+
+alter table "public"."settlements" add constraint "fk_settlements_location" FOREIGN KEY (location_id) REFERENCES locations(id) ON DELETE CASCADE not valid;
+
+alter table "public"."settlements" validate constraint "fk_settlements_location";
+
+alter table "public"."settlements" add constraint "settlements_credit_id_fkey" FOREIGN KEY (credit_id) REFERENCES credits(id) not valid;
+
+alter table "public"."settlements" validate constraint "settlements_credit_id_fkey";
+
+alter table "public"."settlements" add constraint "settlements_debit_id_fkey" FOREIGN KEY (debit_id) REFERENCES debits(id) not valid;
+
+alter table "public"."settlements" validate constraint "settlements_debit_id_fkey";
+
+alter table "public"."settlements" add constraint "settlements_user_id_fkey" FOREIGN KEY (user_id) REFERENCES users(id) not valid;
+
+alter table "public"."settlements" validate constraint "settlements_user_id_fkey";
+
+alter table "public"."tokens" add constraint "tokens_location_id_fkey" FOREIGN KEY (location_id) REFERENCES locations(id) not valid;
+
+alter table "public"."tokens" validate constraint "tokens_location_id_fkey";
+
+alter table "public"."users" add constraint "users_location_id_fkey" FOREIGN KEY (location_id) REFERENCES locations(id) not valid;
+
+alter table "public"."users" validate constraint "users_location_id_fkey";
+
+alter table "public"."users" add constraint "users_wallet_id_fkey" FOREIGN KEY (wallet_id) REFERENCES wallets(id) not valid;
+
+alter table "public"."users" validate constraint "users_wallet_id_fkey";
+
+alter table "public"."wallets" add constraint "new_wallets_wallet_key" UNIQUE using index "new_wallets_wallet_key";
+
+alter table "public"."wallets" add constraint "wallets_address_check" CHECK ((length(address) = 42)) not valid;
+
+alter table "public"."wallets" validate constraint "wallets_address_check";
+
+alter table "public"."wallets" add constraint "wallets_location_id_fkey" FOREIGN KEY (location_id) REFERENCES locations(id) not valid;
+
+alter table "public"."wallets" validate constraint "wallets_location_id_fkey";
+
+set check_function_bodies = off;
+
+CREATE OR REPLACE FUNCTION public.check_null_permit_tx_on_update()
+ RETURNS trigger
+ LANGUAGE plpgsql
+AS $function$BEGIN
+  IF NEW.transaction <> OLD.transaction AND OLD.transaction IS NOT NULL THEN
+    RAISE EXCEPTION 'updating transaction hash is not allowed';
+  END IF;
+
+  RETURN NEW;
+END;$function$
+;
+
+create or replace view "public"."issues_view" as  SELECT DISTINCT ON (locations.issue_id) locations.id,
+    locations.node_id,
+    locations.node_type,
+    locations.updated,
+    locations.created,
+    locations.node_url,
+    locations.user_id,
+    locations.repository_id,
+    locations.organization_id,
+    locations.comment_id,
+    locations.issue_id
+   FROM locations
+  WHERE ((locations.node_url IS NOT NULL) AND (locations.repository_id IS NOT NULL));
+
+
+CREATE OR REPLACE FUNCTION public.read_secret(secret_name text)
```

Comment:
I don't remember this being written by us. Isn't this already defined by Supabase?

Reply:
This one I had added myself, to read secrets inside of Supabase Vault
https://supabase.com/blog/supabase-vault